cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -155,10 +155,13 @@ private String getV3RoutingService(String [] uriParts, AllowedMethod requestMeth
         return Constants.Service.STREAMS;
       }
     } else if ((uriParts.length >= 8 && uriParts[7].equals("logs")) ||
+      (uriParts.length >= 10 && uriParts[9].equals("logs")) ||
```

Comment:
I "love" this router code..... is there a way to do better than this, like by prefix only? Maybe our v3 API is not good?